zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditions.java

```diff
@@ -366,6 +374,62 @@ public static void requireReferredElementsExist(
         });
   }
 
+  /**
+   * This validation applies ONLY to Ad-Hoc Sub-Processes when the {@code #innerInstance} suffix is
+   * present in element IDs. If you're not working with Ad-Hoc Sub-Processes or the {@code
+   * #innerInstance} suffix, you can skip this validation.
```

Comment:
🙃 It's not entirely true. You don't skip it, you fail the validation if it's not an ad-hoc sub-process. In theory it should be impossible as `#`is an invalid character for an element id so it should always be an AHSP.